src/examples/echo_server.c

```diff
@@ -0,0 +1,107 @@
+#include <mtev_defines.h>
+#include <mtev_conf.h>
+#include <mtev_console.h>
+#include <mtev_dso.h>
+#include <mtev_listener.h>
+#include <mtev_main.h>
+#include <mtev_memory.h>
+#include <mtev_http.h>
+#include <mtev_rest.h>
+#include <mtev_cluster.h>
+#include <mtev_capabilities_listener.h>
+#include <mtev_events_rest.h>
+#include <eventer/eventer.h>
+
+#include <errno.h>
+#include <stdio.h>
+#include <getopt.h>
+
+#define APPNAME "echo_server"
+static char *config_file = NULL;
+static int debug = 0;
+static int foreground = 0;
+static char *droptouser = NULL, *droptogroup = NULL;
+
+static int
+usage(const char *prog) {
+	fprintf(stderr, "%s <-c conffile> [-D] [-d]\n\n", prog);
+  fprintf(stderr, "\t-c conffile\tthe configuration file to load\n");
+  fprintf(stderr, "\t-D\t\trun in the foreground (don't daemonize)\n");
+  fprintf(stderr, "\t-d\t\tturn on debugging\n");
+  return 2;
+}
+static void
+parse_cli_args(int argc, char * const *argv) {
+  int c;
+  while((c = getopt(argc, argv, "c:Dd")) != EOF) {
+    switch(c) {
+      case 'c':
+        config_file = optarg;
+        break;
+      case 'd': debug = 1; break;
+      case 'D': foreground++; break;
+    }
+  }
+}
+
+static int my_post_handler(mtev_http_rest_closure_t *restc, int npats, char **pats) {
+
+  size_t buffer_size = 128 * 1024;
+  char *buffer = alloca(buffer_size);
+  int mask = EVENTER_READ | EVENTER_WRITE | EVENTER_EXCEPTION;
+  int len;
+  int done = 0;
+
+  while (!done) {
+    len = mtev_http_session_req_consume(restc->http_ctx, buffer, buffer_size, buffer_size, &mask);
+    if (len > 0) {
+      mtev_http_response_append(restc->http_ctx, buffer, len);    
+    }
+    if ((len < 0 && errno != EAGAIN) || len == 0) {
+      done = 1;
+    }
+  } 
+  
+  mtev_http_response_standard(restc->http_ctx, 200, "OK", "text/plain");
+  mtev_http_response_header_set(restc->http_ctx, "Content-Type", "text/plain");
+  mtev_http_response_option_set(restc->http_ctx, MTEV_HTTP_CLOSE);
+  mtev_http_response_option_set(restc->http_ctx, MTEV_HTTP_LZ4F);
+  mtev_http_response_end(restc->http_ctx);
```

Comment:
I thought all the response headers needed to be set before we started appending the response body?